### PR TITLE
Added Webhook object

### PR DESCRIPTION
**Pull request by zshannon**
We wanted to interact with received Webhook events in an object oriented way, so we modeled the Webhook object.

### CODE REVIEW

File: lib/plaid/webhook.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
+module Plaid
+  # Public: Representation of a webhook.
+  class Webhook
+    # Public: The String human readable explanation of this webhook request.
+    # E.g. "Initial transaction pull finished".
+    attr_reader :message
+
+    # Public: The access token for authenticated user.
```
Comment (be9) [inline lib/plaid/webhook.rb]: The **String** access token ...?

File: lib/plaid/webhook.rb [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      @removed_transactions = fields['removed_transactions']
+      # present only for Error Response Webhook
+      @resolve = fields['resolve']
+    end
+
+    # Public: Share the type of Webhook this is from its code
+    #
+    # Returns String webhook type
+    def type
+      Webhook::CODEX[@code] || 'ErrorResponse'
```
Comment (be9) [inline lib/plaid/webhook.rb]: Maybe just use `code` instead of `@code`, since you have `attr_reader :code`?

File: lib/plaid/webhook.rb [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    end
+
+    # Public: Get a String representation of the transaction.
+    #
+    # Returns a String.
+    alias to_s inspect
+
+    private
+
+    Webhook::InitialTransaction    = 0
```
Comment (be9) [inline lib/plaid/webhook.rb]: I'd use ALL_CAPS for constants here. There's also duplication which can be avoided by using metaprogramming:

```ruby
codex = {}
['initial transaction', 'historical transaction'].each_with_index do |event, idx|
  name = event.split.map(&:upcase).join('_')
  codex[idx + 1] = name  
  const_set name, idx + 1
end

CODEX = codex.freeze
```

**Comment by be9** [comment]
Thanks @zshannon! See my review comment.

**Comment by zshannon** [comment]
@be9 Thanks for the quick and careful review. The changes you requested are here now.

File: lib/plaid/webhook.rb [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      @removed_transactions = fields['removed_transactions']
+      # present only for Error Response Webhook
+      @resolve = fields['resolve']
+    end
+
+    # Public: Share the type of Webhook this is from its code
+    #
+    # Returns String webhook type
+    def type
+      Webhook::CODEX[code] || 'ErrorResponse'
```
Comment (be9) [inline lib/plaid/webhook.rb]: `'ERROR_RESPONSE'`?

Comment (zshannon) [inline lib/plaid/webhook.rb]: Fixed up.

**Comment by be9** [comment]
Thanks for your contribution @zshannon ! 👍

It has just been released as part of 2.1.0.

**Pull request merged**